### DISCUSSION TITLE
is JavaMelody affected by CVE-2021-44228?

### DISCUSSION THREAD

**Discussion by bertorga** [javamelody/javamelody#1059] [General]
Hello,
Do you know if JavaMelody is affected by CVE-2021-44228?
I've been looking up but I found nothing.
Thank you very much

**Comment by tjj225** [comment]
Java melody CORE is not affected by this cve. In general, it does not include log4j as the dependency is [optional](https://github.com/javamelody/javamelody/blob/8aae405ec88fd71cb92346b8c3e1857ff3e32065/javamelody-core/pom.xml#L108).

**Reply by bertorga** [reply]
Thank you very much!!

**Comment by evernat** [comment]
Most applications monitored by javamelody use either the dependency (jar file) javamelody-core or the dependency javamelody-spring-boot-starter. Both are not affected by the CVE, because they do not include log4J and they do not declare a compile or runtime dependency on log4j.

Other applications use a plugin for Jenkins or for JIRA/Confluence/Bamboo/Bitbucket or for Liferay or for Alfresco or for Sonar or for Grails. Those plugins are not affected for the same reason.

The optional javamelody collect server is sometimes used, even if much less than javamelody-core or javamelody-spring-boot-starter or the javamelody plugins. When used, it is in a different server (that is not in the monitored applications).
The optional javamelody collect server includes log4j, for which the recent security advisories were published (CVE-2021-44228 and CVE-2021-45046).
So when the collect server is used, it is recommended to upgrade the collect server to the latest version 1.90.0, which includes the latest log4j 2.16.0.
You can find the version 1.90.0 of the collect server at https://github.com/javamelody/javamelody/releases/download/javamelody-core-1.90.0/javamelody-collector-server-1.90.0.war
